synapse/util/caches/lrucache.py

```diff
@@ -241,8 +282,16 @@ def cache_set_default(key, value):
                 evict()
                 return value
 
+        @overload
+        def cache_pop(key: KT, default: Literal[None] = None) -> Union[None, VT]:
```

Comment:
I think this is equivalent?

```suggestion
        def cache_pop(key: KT, default: Literal[None] = None) -> Optional[VT]:
```